hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -115,6 +115,22 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
           OMException.ResultCodes.INVALID_REQUEST);
     }
 
+    // If this is a Link Bucket, we make sure that its layout matches the
+    //  source bucket. This prevents inconsistencies during list operations.
+    if (hasSourceBucket) {
+      try {
+        newBucketInfo.setBucketLayout(
```

Comment:
Still there is one gap for the pre-created bucket-links. Those buckets will mismatched with the real source bucket. Instead of adding a logic here, how about checking `omBucketInfo#isLink()` during listing bucket then get the source bucket's layout, if source exists?

https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java#L826

Reply:
I have added this in `a572156`. Currently the fix is only implemented for RpcClient#listBuckets. I'm working to get it working for OmMetadataManagerImpl#listBuckets.